apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/common/db/SearchSpecification.java

```diff
@@ -0,0 +1,52 @@
+package edu.pw.apsienrollment.common.db;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.Path;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import java.util.Arrays;
+import java.util.Collection;
+
+@RequiredArgsConstructor
+public abstract class SearchSpecification<S> implements Specification<S> {
+    protected final Collection<SearchCriteria> searchCriteria;
+
+    protected Predicate toCommonSearchPredicate(SearchCriteria criteria, Root<S> root, CriteriaBuilder criteriaBuilder) {
+        Path<String> path = resolvePath(criteria.getKey(), root);
+        if (criteria.getOperation().equalsIgnoreCase(">")) {
```

Comment:
`ignoreCase` is not required for given operators, as they are nonliteral characters